options/options.go

```diff
@@ -240,3 +250,44 @@ func validateFormat(format string) bool {
 		return false
 	}
 }
+
+// isValidForLocalRepo returns an error if the check is not valid for local repository.
+func (o *Options) isValidForLocalRepo() error {
+	if o.Local == "" {
+		return nil
+	}
+	var requiredRequestTypes []checker.RequestType
+	requiredRequestTypes = append(requiredRequestTypes, checker.FileBased)
+
+	for _, check := range o.ChecksToRun {
+		if !policy.IsSupportedCheck(check, requiredRequestTypes) {
+			// Too many errors need to be created to satisfy the linter.
+			//nolint
+			return fmt.Errorf(
+				"check %s is not supported for local repositories",
+				check,
+			)
+		}
+	}
+	return nil
+}
+
+// isValidForNonLocalRepo returns an error if the check is not valid for non-local repository.
+func (o *Options) isValidForNonLocalRepo() error {
```

Comment:
Maybe rename to `isValidForNonLocalGitRepo`. I initially thought this meant remote repos for checks that require GitHub APIs